File: codesamples/old-samples/sample.js
// define interactionSignal handler in Express, Next.js, Hapi, Koa, Fastify, etc...
export async function mainWorkflow(userId) {
  const userInteracted = await Promise.race([
    interactionSignal, // receive signals from users!
    sleep(30 * DAY) // sleep for 30 days!
  ]);
  if (!userInteracted) {
    await sendEmailActivity(userId); // timeouts + retries!
  }
}
// Scale to Millions + Write Tests + Encrypt Data + Migrate Versions + ...

File: codesamples/old-samples/sample.go
func mainWorkflow(ctx workflow.Context, userId) error {
		// ...
		selector.AddReceive(signalChannel, func(c workflow.ReceiveChannel, more bool) {
				c.Receive(ctx, &signalVal) // receive user signals!
		})
		timerFuture := workflow.NewTimer(childCtx, 30 * time.Day) // sleep for 30 days!
		selector.AddFuture(timerFuture, func(f workflow.Future) {
				_ = workflow.ExecuteActivity(ctx, SendEmailActivity).Get(ctx, nil) // timeouts and retries!
		})
		// Scale to Millions + Write Tests + Encrypt Data + Migrate Versions + ... 
}